branch-validation.yml

```diff
@@ -183,26 +183,30 @@
       set_fact: master=true
       when: floating_ip in ansible_all_ipv4_addresses
 
-    # This is manual for now, not keen on automated kernel panic
+    # initiate hard reboot on the current master controller
     - name: initiate a failover on controller 0
-      pause: prompt="On ucarp {{ inventory_hostname }} $ echo c > /proc/sysrq-trigger (press enter when done)"
+      shell: echo 'b' | tee /proc/sysrq-trigger
       when: master is defined
 
     - name: initiate a failover on controller 1
-      pause: prompt="On ucarp {{ groups['controller'][1] }} $ echo c > /proc/sysrq-trigger (press enter when done)"
+      delegate_to: "{{ groups['controller'][1] }}"
+      shell: echo 'b' | tee /proc/sysrq-trigger
       when: master is not defined
 
     - name: wait for the panic
       wait_for: host={{ inventory_hostname }} port=22 state=stopped delay=10
                 timeout=30
-      delegate_to: localhost
       when: master is defined
+      delegate_to: localhost
 
     - name: wait for the panic
       wait_for: host={{ groups['controller'][1] }} port=22 state=stopped
                 delay=10 timeout=30
-      delegate_to: localhost
       when: master is not defined
+      delegate_to: localhost
+
+    - name: wait for the node to boot
+      pause: minutes=3
```

Comment:
oh, I see what's going on here. The wait_for is already waiting for ssh to stop responding, but even then ping would respond. I get what you're saying. 3 minutes here might be too long, is there a chance the host would be already coming back up by then?